ftw/file/tests/test_pagination.py

```diff
@@ -0,0 +1,42 @@
+from ftw.builder import Builder
+from ftw.builder import create
+from ftw.file.testing import FTW_FILE_FUNCTIONAL_TESTING
+from ftw.testbrowser import browsing
+from plone.app.testing import setRoles
+from plone.app.testing import TEST_USER_ID
+from unittest2 import TestCase
+import transaction
+
+
+class TestPagination(TestCase):
+    layer = FTW_FILE_FUNCTIONAL_TESTING
+
+    def setUp(self):
+        self.portal = self.layer['portal']
+        setRoles(self.layer['portal'], TEST_USER_ID, ['Contributor'])
+
+        portal_repository = self.portal.portal_repository
+        portal_repository.addPolicyForContentType(u'File',
+                                                  u'at_edit_autoversion')
+        portal_repository.setVersionableContentTypes(
+            portal_repository.getVersionableContentTypes() + [u'File'])
+        transaction.commit()
+
+        self.file = create(Builder('file')
+                           .titled('An important file')
+                           .with_dummy_content())
+
+    @browsing
+    def test_pagination_is_used(self, browser):
+        for i in range(14):
```

Comment:
I define the batch size in the template. So i don't think i can change it in the test. But i check how to fill the history manually atm.